nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -10,14 +10,23 @@ let
   cfg = config.services.mastodon;
   opt = options.services.mastodon;
 
+  # This should probably live in pkgs.formats, but pkgs.formats.keyValue doesn't suit our usecase yet.
+  settingToEnvVar =
```

Comment:
Isn't the first attribute-set passed to generators exactly for this use-case? I.e. to change certain aspects of the generator? Or am I misunderstanding the problem?

Reply:
Yes it is, but currently none of the generators allow the behaviour we want here. Eventually we'd want something like `pkgs.formats.envVars` that is customisable with a `[a] -> str` function and is well suited to be plugged into `systemd.services.<>.environment`.

Reply:
I did my own rfc-42-ification of systemd recently (https://github.com/NixOS/nixpkgs/pull/426692), and emily made some suggestions about potentially reworking systemd environment variable serialisation. But that would be its own PR, and more part of systemd support in nixpkgs, not something relevant to mastodon specifically. And probably wouldn't be a `pkgs.formats`, rather something as part of the special-casing systemd does for its ini-style config files that are already aware of lists.

Reply:
fwiw, the formats lib has a `toGitINI`. For stuff that's sufficiently common in nixpkgs (systemd-style INI or whatever we want to call it does count here IMHO), I'd argue that it's OK to do the same.